### PR TITLE
fix(web): add comment cya page tweaks

### PR DESCRIPTION
**Pull request by Si3rr4wow**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: appeals/web/src/server/appeals/appeal-details/interested-party-comments/add-ip-comment/add-ip-comment.controller.js [hunk 1]
```diff
@@ -239,20 +246,104 @@ export async function postIPComment(request, response) {
...
+				firstName: '',
+				lastName: '',
+				emailAddress: '',
+				addressProvided: '',
+				addressLine1: '',
+				addressLine2: '',
+				town: '',
+				county: '',
+				postCode: ''
+			}
```
Comment (Si3rr4wow) [inline appeals/web/src/server/appeals/appeal-details/interested-party-comments/add-ip-comment/add-ip-comment.controller.js:286]: Not keen on this but there doesn't seem to be a way to mock the session in the tests so setting defaults was the best I could come up with to make them pass

File: appeals/web/src/server/appeals/appeal-details/interested-party-comments/add-ip-comment/__tests__/__snapshots__/add-ip-comment.test.js.snap [hunk 2]
```diff
@@ -134,25 +134,24 @@ exports[`add-ip-comment GET /check-your-answers should match the snapshot 1`] =
...
                     </dd>
                     <dd class="govuk-summary-list__actions"><a class="govuk-link" href="/appeals-service/appeal-details/2/interested-party-comments/add/upload"> Change</a>
                     </dd>
                 </div>
-                <div class="govuk-summary-list__row"><dt class="govuk-summary-list__key"> Redaction Status</dt>
-                    <dd class="govuk-summary-list__value"></dd>
-                    <dd class="govuk-summary-list__actions"><a class="govuk-link" href="/appeals-service/appeal-details/2/interested-party-comments/add/redaction-status"> Change</a>
+                <div class="govuk-summary-list__row"><dt class="govuk-summary-list__key"> Redaction status</dt>
+                    <dd class="govuk-summary-list__value">Unredacted</dd>
+                    <dd class="govuk-summary-list__actions"><a class="govuk-link" href="/appeals-service/appeal-details/2/interested-party-comments/undefined/add-document/redaction-status"> Change</a>
```
Comment (james-hill-kc) [inline appeals/web/src/server/appeals/appeal-details/interested-party-comments/add-ip-comment/__tests__/__snapshots__/add-ip-comment.test.js.snap:144]: There's an `undefined` in the URL here

File: appeals/web/src/server/appeals/appeal-details/interested-party-comments/add-ip-comment/add-ip-comment.controller.js (hunk 1 above)
Comment (james-hill-kc) [reply appeals/web/src/server/appeals/appeal-details/interested-party-comments/add-ip-comment/add-ip-comment.controller.js:286]: I agree this is pretty unreadable. If it's just for the sake of tests, my preference would be to remove this and simply not test whatever becomes untestable as a result (as you say, there's no way to mock the session as far as I know)

File: appeals/web/src/server/appeals/appeal-details/interested-party-comments/add-ip-comment/add-ip-comment.controller.js [hunk 3]
```diff
@@ -239,20 +246,104 @@ export async function postIPComment(request, response) {
...
+		throw new Error('Received invalid redaction status');
+	}
 
-	return response
-		.status(request.errors ? 400 : 200)
-		.render('patterns/check-and-confirm-page.pattern.njk', {
-			errors: request.errors,
-			pageContent
-		});
+	return renderCheckYourAnswersComponent(
```
Comment (james-hill-kc) [inline appeals/web/src/server/appeals/appeal-details/interested-party-comments/add-ip-comment/add-ip-comment.controller.js:295]: The problem with moving this out of the mapper into the controller is you're losing the separation of concerns and encapsulation of logic. This function is so generic you basically have to pass the entire contents of the old mapper function as parameters, at which point you may as well not have the helper and just define the CYA page components inline in the controller code.

**Review by james-hill-kc — COMMENTED** [review]
I've left a couple of comments, overall I'm not keen on these changes. The ticket seems to only be about copy updates, but there's a lot of refactoring in here that seems unrelated, and this also seems to have broken some of the action link URLs.

**Review by glpins — APPROVED** [review]

**Pull request merged**